exercises/diffie-hellman/example.rs

```diff
@@ -0,0 +1,18 @@
+extern crate rand;
+
+use rand::distributions::{IndependentSample, Range};
+
+pub fn private_key(p: &u64) -> u64 {
```

Comment:
The `u64` type is `Copy`, so it's more idiomatic and convenient to pass by value instead of by reference. 

Please update the tests and the example such that all `u64`s are passed by value instead of by reference.